xdsl/backend/csl/lowering/convert_stencil_to_csl.py

```diff
@@ -0,0 +1,399 @@
+from attr import dataclass
+
+from xdsl.dialects import func
+from xdsl.dialects.arith import Cmpi, Constant, MinUI, Muli, OrI
+from xdsl.dialects.builtin import (
+    IntegerAttr,
+    IntegerType,
+    MemRefType,
+    ModuleOp,
+    Signedness,
+    StringAttr,
+    TensorType,
+)
+from xdsl.dialects.csl import (
+    ComptimeStructType,
+    ConcatStructOp,
+    ConstStructOp,
+    CslModuleOp,
+    GetColorOp,
+    ImportModuleConstOp,
+    LayoutOp,
+    MemberCallOp,
+    ModuleKind,
+    ModuleKindAttr,
+    ParamOp,
+    SetRectangleOp,
+    SetTileCodeOp,
+    SignednessCastOp,
+)
+from xdsl.dialects.scf import For, Yield
+from xdsl.dialects.stencil import ApplyOp
+from xdsl.ir import Attribute, Block, MLContext, Operation, Region
+from xdsl.passes import ModulePass
+from xdsl.utils.hints import isa
+from xdsl.utils.str_enum import StrEnum
+
+
+class Named(StrEnum):
+    """
+    Enum type to store names used across the program, for instance
+    names of parameters passed from the layout to the program module.
+    """
+
+    pattern = ("pattern",)
+    stencil_comms_params = ("stencil_comms_params",)
+    memcpy_params = ("memcpy_params",)
+    z_dim = ("z_dim",)
+    is_border_region_pe = ("is_border_region_pe",)
+
+
+class TranslationContext:
```

Comment:
I feel like this would benefit from a `@dataclass` annotation

Reply:
good point

Reply:
But then I can't do a standard dict access to the fields..